src/lib.rs

```diff
@@ -4,36 +4,47 @@ mod toolchain;
 use std::collections::HashMap;
 use std::fs::{create_dir_all, write};
 use std::path::{Path, PathBuf};
-use std::str::FromStr;
 
 use anyhow::{bail, Result};
+use arg_enum_proc_macro::ArgEnum;
 use minijinja::value::Value;
 use minijinja::{Environment, Source};
 
 use filters::*;
 use toolchain::*;
 
 /// Supported templates
-enum Templates {
+#[derive(ArgEnum, Debug)]
+pub enum Templates {
+    /// Generate a meson project using C as main language
+    #[arg_enum(name = "meson-c")]
     MesonC,
+    /// Generate a meson project using C++ as main language
+    #[arg_enum(name = "meson-c++")]
     MesonCpp,
-    CargoCI,
+    /// Generate a pyproject.toml project using Python as main language
+    #[arg_enum(name = "setuptools")]
     SetupTools,
+    /// Generate a Github Action and Gitlab-CI setup for a cargo project
+    #[arg_enum(name = "cargo-ci")]
+    CargoCI,
+    /// Generate a Github Action and Gitlab-CI setup for a yarn project
+    #[arg_enum(name = "yarn-ci")]
     YarnCI,
 }
 
-impl FromStr for Templates {
-    type Err = ();
-
-    fn from_str(input: &str) -> Result<Self, Self::Err> {
-        match input {
-            "meson-c" => Ok(Self::MesonC),
-            "meson-c++" => Ok(Self::MesonCpp),
-            "cargo-ci" => Ok(Self::CargoCI),
-            "setuptools" => Ok(Self::SetupTools),
-            "yarn-ci" => Ok(Self::YarnCI),
-            _ => Err(()),
+impl Templates {
+    pub fn info() -> String {
+        let mut info = "Available built-in templates:\n".to_string();
+        for (names, description) in Templates::descriptions() {
+            std::fmt::write(
+                &mut info,
+                format_args!("{:<15} {}\n", names[0], description[0]),
+            )
+            .unwrap();
```

Comment:
Do not use `unwrap` inside the lib, let's return a `Result`

Reply:
It cannot fail.

Reply:
Yeah, but the use of an `unwrap` could be contraindicated in a lib in my opinion, anyway is just a nit